code/model/modelanim.cpp

```diff
@@ -868,7 +868,8 @@ void model_anim_set_initial_states(ship *shipp)
 			if (psub->triggers[i].type == AnimationTriggerType::Initial) {
 				if (psub->type == SUBSYSTEM_TURRET) {
 					// special case for turrets
-					pss->submodel_instance_2->angs.p = psub->triggers[i].angle.xyz.x;
+					if (pss->submodel_instance_2 != nullptr)
+						pss->submodel_instance_2->angs.p = psub->triggers[i].angle.xyz.x;
 					pss->submodel_instance_1->angs.h = psub->triggers[i].angle.xyz.y;
```

Comment:
I see you added null checks for the `_1` variant below so is that also necessary here?

Reply:
In the scripting case, a script might call the functions on any old subsystem, whether it's a turret or not.  In this case, I would assume upstream checks should check this before we reach this point.  However, that's an assumption, so I think a null check does make sense.